### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Hjorthmedh**

### CODE REVIEW

File: examples/notebooks/arvind_dendrite_role/generate_network2.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+export IPYTHONDIR="`pwd`/.ipython"
```
Comment (robban80) [inline examples/notebooks/arvind_dendrite_role/generate_network2.sh:1]: is this file really necessary? It is basically a duplicate of the original file (I suspect this will clutter the directory)

File: examples/notebooks/validation/neuromodulation/neuromodulation_sim.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 def analyse(network_path):
```
Comment (robban80) [inline examples/notebooks/validation/neuromodulation/neuromodulation_sim.py:9]: it is not clear why this update is done and what it is doing

Comment (robban80) [reply examples/notebooks/validation/neuromodulation/neuromodulation_sim.py:9]: i.e. why iSPN is removed from the validation

File: snudda/init/init.py [hunk 3]
```diff
@@ -837,10 +836,11 @@ def define_striatum(self,
                         mesh_file=None,
                         mesh_bin_width=None,
                         d_min=None,
-                        cluster_FS_synapses=False,
+                        cluster_FS_synapses=False,       # !!! TEMP SET TO TRUE
```
Comment (robban80) [inline snudda/init/init.py:839]: remove comment?

File: snudda/init/init.py [hunk 4]
```diff
@@ -1346,30 +1346,32 @@ def define_striatum(self,
         # ================================================================
```
Comment (robban80) [inline snudda/init/init.py:1346]: No reason to keep these comments. I suggest to remove the if statement and the comments

Comment (robban80) [reply snudda/init/init.py:1346]: also the conductance is set to ChINgGABA while the type is set to ACh - what will this do?

File: snudda/simulate/simulate.py [hunk 5]
```diff
@@ -897,6 +901,10 @@ def add_synapse(self, cell_id_source, cell_id_dest, dend_compartment, section_id
 
         (channel_module, par_data) = self.synapse_parameters[synapse_type_id]
 
+        if channel_module is None:
```
Comment (robban80) [inline snudda/simulate/simulate.py:904]: perhaps a warning would be in place here?

File: snudda/utils/load_network_simulation.py [hunk 6]
```diff
@@ -161,51 +166,71 @@ def get_voltage(self, neuron_id=None):
         else:
             return voltage
 
-    """
-    def check_depolarisation_block(self, threshold=0, max_duration=100e-3):
```
Comment (robban80) [inline snudda/utils/load_network_simulation.py:165]: I suggest that the duration is lowered here. Some anomalies are not lasting 50 ms and would not be detected. I use 20 ms with good results

File: snudda/utils/swap_to_degenerated_morphologies.py [hunk 7]
```diff
@@ -74,10 +74,17 @@ def write_new_network_file(self):
...
+        if len(self.new_snudda_data_dir) > len(self.original_snudda_data_dir):
+            del self.new_hdf5["meta/snuddaData"]
+            self.new_hdf5["meta"].create_dataset("snuddaData", data=self.new_snudda_data_dir)
+        else:
+            self.new_hdf5["meta/SnuddaData"] = self.new_snudda_data_dir
         network_group = self.new_hdf5.create_group("network")
         self.old_hdf5.copy(source=self.old_hdf5["network/neurons"], dest=self.new_hdf5["network"])
 
-        if len(self.new_snudda_data_dir) > len(self.original_snudda_data_dir):
+        # This if statement is now redundant, since we express neuron paths using SNUDDA_DATA
```
Comment (robban80) [inline snudda/utils/swap_to_degenerated_morphologies.py:86]: Perhaps this should be marked for deletion somehow?

File: snudda/utils/swap_to_degenerated_morphologies_extended.py [hunk 8]
```diff
@@ -165,27 +221,39 @@ def filter_synapses(self, filter_axon=False):
...
         # Resort the new synapse matrix
         sorted_synapses = self.sort_synapses(new_synapses[:syn_ctr, :])
 
         old_synapse_iterator = self.synapse_iterator(synapses=self.old_hdf5["network/synapses"][()])
-        config = json.loads(self.updated_network_loader.data["config"], object_pairs_hook=OrderedDict)
 
-        pruned_synapses = self.post_degeneration_pruning(synapses=sorted_synapses,
-                                                         old_synapse_iterator=old_synapse_iterator,
-                                                         network_config=config,
-                                                         rng=self.rng)
```
Comment (robban80) [inline snudda/utils/swap_to_degenerated_morphologies_extended.py:186]: I suggest to remove the if-else statement here

**Review by robban80 — COMMENTED** [review]
I have gone trough the changes and left some minor comments. However I'm not sure I would spot any major flaw here since I'm not familiar enough with the code...

File: snudda/init/init.py (hunk 3 above)
Comment (Hjorthmedh) [reply snudda/init/init.py:839]: Removing.

File: snudda/init/init.py (hunk 4 above)
Comment (Hjorthmedh) [reply snudda/init/init.py:1346]: Removed

Comment (Hjorthmedh) [reply snudda/init/init.py:1346]: Changed to ChINgACh

File: snudda/simulate/simulate.py (hunk 5 above)
Comment (Hjorthmedh) [reply snudda/simulate/simulate.py:904]: Adding. Had it before, but there were so many printouts. Now added logic so that it will only print error once.

File: snudda/utils/load_network_simulation.py (hunk 6 above)
Comment (Hjorthmedh) [reply snudda/utils/load_network_simulation.py:165]: Changed.

File: snudda/utils/swap_to_degenerated_morphologies.py (hunk 7 above)
Comment (Hjorthmedh) [reply snudda/utils/swap_to_degenerated_morphologies.py:86]: Removed

File: snudda/utils/swap_to_degenerated_morphologies_extended.py (hunk 8 above)
Comment (Hjorthmedh) [reply snudda/utils/swap_to_degenerated_morphologies_extended.py:186]: Is this comment supposed to refer to another line?

File: examples/notebooks/arvind_dendrite_role/generate_network2.sh (hunk 1 above)
Comment (Hjorthmedh) [reply examples/notebooks/arvind_dendrite_role/generate_network2.sh:1]: Removing

File: examples/notebooks/validation/neuromodulation/neuromodulation_sim.py (hunk 2 above)
Comment (Hjorthmedh) [reply examples/notebooks/validation/neuromodulation/neuromodulation_sim.py:9]: The validation directory is the validations against experimental data, pipeline is being set up by Shailesh

File: snudda/utils/swap_to_degenerated_morphologies_extended.py (hunk 8 above)
Comment (Hjorthmedh) [reply snudda/utils/swap_to_degenerated_morphologies_extended.py:186]: found it
